src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,171 @@
+package duke.util;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.format.DateTimeParseException;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.InvalidCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.UnmarkCommand;
+import duke.exception.DukeException;
+import duke.exception.InvalidCommandFormatException;
+import duke.exception.InvalidDateException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * This class parses the user input into commands and also parses the saved task list on disk into tasks.
+ */
+public class Parser {
+
+    // Global constant commands.
+    private static final SimpleDateFormat taskFormat = new SimpleDateFormat("MMM dd yyyy");
```

Comment:
Shouldn't constant names be all uppercase using underscore to separate words?

Reply:
Yep changed it to `TASK_FORMAT` and `SAVED_FORMAT`.